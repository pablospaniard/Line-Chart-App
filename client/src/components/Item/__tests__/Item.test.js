import React from 'react'
import { render, cleanup, waitForElement } from 'react-testing-library'

import { Item } from '..'
import data from '../../Layout/__tests__/data.mock.json'

describe('Item should works properly', () => {
  const fetchData = jest.fn()

  afterEach(cleanup)

  it('should match snapshot', () => {
    const { container } = render(
      <Item dataSet={data.data} fetchData={fetchData} />
    )
    expect(container.firstChild).toMatchSnapshot('Item_snapshot_1')
  })

  it('should show Chart if loading is false', async () => {
    const { getByTestId } = render(
      <Item dataSet={data.data} fetchData={fetchData} loading={false} />
    )
    await waitForElement(() => getByTestId('chart'))
  })

  it('should show Spinner if loading is true', async () => {
    const { getByTestId } = render(
      <Item dataSet={data.data} fetchData={fetchData} loading={true} />
    )
    await waitForElement(() => getByTestId('spinner'))
  })
})
