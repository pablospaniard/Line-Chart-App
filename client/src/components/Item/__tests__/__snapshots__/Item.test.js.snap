// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should works properly should match snapshot: Item_snapshot_1 1`] = `
<div
  class="MuiGrid-container-1 Item"
>
  <div
    class="MuiGrid-item-2 MuiGrid-grid-xs-6-35 Chart"
    data-testid="chart"
  >
    <canvas
      height="150"
      width="300"
    />
  </div>
</div>
`;
